src/random_ray/flat_source_domain.cpp

```diff
@@ -432,11 +437,61 @@ void FlatSourceDomain::convert_source_regions_to_tallies()
 // Set the volume accumulators to zero for all tallies
 void FlatSourceDomain::reset_tally_volumes()
 {
+  if (volume_normalized_flux_tallies_) {
 #pragma omp parallel for
-  for (int i = 0; i < tally_volumes_.size(); i++) {
-    auto& tensor = tally_volumes_[i];
-    tensor.fill(0.0); // Set all elements of the tensor to 0.0
+    for (int i = 0; i < tally_volumes_.size(); i++) {
+      auto& tensor = tally_volumes_[i];
+      tensor.fill(0.0); // Set all elements of the tensor to 0.0
+    }
+  }
+}
+
+// In fixed source mode, due to the way that volumetric fixed sources are
+// converted and applied as volumetric sources in one or more source regions,
+// we need to perform an additional normalization step to ensure that the
+// reported scalar fluxes are in units per source neutron. This allows for
+// direct comparison of reported tallies to Monte Carlo flux results.
+double FlatSourceDomain::compute_fixed_source_normalization_factor() const
+{
+  // If we are not in fixed source mode, then there are no external sources
+  // so no normalization is needed.
+  if (settings::run_mode != RunMode::FIXED_SOURCE) {
+    return 1.0;
+  }
+
+  // Step 1 is to sum over all source regions and energy groups to get the
+  // total external source strength in the simulation.
+  double simulation_external_source_strength = 0.0;
+#pragma omp parallel for reduction(+ : simulation_external_source_strength)
+  for (int sr = 0; sr < n_source_regions_; sr++) {
+    int material = material_[sr];
+    double volume = volume_[sr] * simulation_volume_;
+    for (int e = 0; e < negroups_; e++) {
+      // Temperature and angle indices, if using multiple temperature
+      // data sets and/or anisotropic data sets.
+      // TODO: Currently assumes we are only using single temp/single
+      // angle data.
+      const int t = 0;
+      const int a = 0;
+      float sigma_t = data::mg.macro_xs_[material].get_xs(
+        MgxsType::TOTAL, e, nullptr, nullptr, nullptr, t, a);
+      simulation_external_source_strength +=
+        external_source_[sr * negroups_ + e] * sigma_t * volume;
```

Comment:
Since we are using `sr` as the index variable for **s**ource **r**egions, I think it makes sense to use `eg` as the index variable for **e**nergy **g**roup
```suggestion
    for (int eg = 0; eg < negroups_; eg++) {
      // Temperature and angle indices, if using multiple temperature
      // data sets and/or anisotropic data sets.
      // TODO: Currently assumes we are only using single temp/single
      // angle data.
      const int t = 0;
      const int a = 0;
      float sigma_t = data::mg.macro_xs_[material].get_xs(
        MgxsType::TOTAL, eg, nullptr, nullptr, nullptr, t, a);
      simulation_external_source_strength +=
        external_source_[sr * negroups_ + eg] * sigma_t * volume;
```

Reply:
This brings up a good point! As it is in the random ray solver, we currently have `e` and `g` used as iterators over energy groups. The `eg` idea is also reasonable. I think it would be best though to pick one and be consistent with it throughout random ray source files in OpenMC, and to add it to the source style guide at https://docs.openmc.org/en/stable/devguide/styleguide.html#naming

My vote would be to use `g` consistently for these loops, as the MGMC portion of the code is already using `gin` and `gout` for incoming/outgoing energy groups. Additionally, `g` matches the verbiage in all the random ray equations in the docs, which would be more of a pain to refactor as compared to the source code itself.

Reply:
I would propose that the variable names be left as they are in this PR, and I can commit to a separate PR that does the refactor on energy group loop variables for consistency throughout.

Reply:
`eg` is the mots clear to me, but `g` also makes sense. I agree, we should pick one and make it consistent across the whole codebase and add it to the source style guide. Doing all of that in a separate PR would be great!